building/build-debs/homeworld-keysystem/src/util/fileutil/util.go

```diff
@@ -0,0 +1,57 @@
+package fileutil
+
+import (
+	"errors"
+	"fmt"
+	"os"
+)
+
+func Exists(path string) bool {
+	f, err := os.Open(path)
+	if err == nil {
+		f.Close()
+		return true
+	} else {
+		return false
+	}
+}
+
+func EnsureIsFolder(dirname string) error {
+	fileinfo, err := os.Stat(dirname)
+	if err != nil {
+		if os.IsNotExist(err) {
+			err = os.MkdirAll(dirname, 0755)
+		}
+		if err != nil {
+			return err
+		}
+		fileinfo, err = os.Stat(dirname)
+		if err != nil {
+			return err // hard to test in practice, but still necessary
+		}
+	}
+	if !fileinfo.IsDir() {
+		return errors.New("not a directory")
+	}
+	return nil
+}
+
```

Comment:
Anything using these functions suffers from TOCTTOU.  It's not a problem if we always have our own isolated filesystem, I guess?

Reply:
Nothing in these functions is designed for security -- the files accessed are generally under places like /etc/homeworld/, and anyone with write access there can already gain full access to everything the keyclient has access to, because they can just read the private keys it has.